units/event_metajob.go

```diff
@@ -0,0 +1,202 @@
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/sometimes"
+	"github.com/pkg/errors"
+)
+
+const (
+	eventMetaJobName = "event-metajob"
+)
+
+func init() {
+	registry.AddJobType(eventMetaJobName, func() amboy.Job { return makeEventMetaJob() })
+}
+
+func notificationIsEnabled(flags *evergreen.ServiceFlags, n *notification.Notification) bool {
+	switch n.Subscriber.Type {
+	case event.GithubPullRequestSubscriberType:
+		return !flags.GithubStatusAPIDisabled
+
+	case event.JIRAIssueSubscriberType, event.JIRACommentSubscriberType:
+		return !flags.JIRANotificationsDisabled
+
+	case event.EvergreenWebhookSubscriberType:
+		return !flags.WebhookNotificationsDisabled
+
+	case event.EmailSubscriberType:
+		return !flags.EmailNotificationsDisabled
+
+	case event.SlackSubscriberType:
+		return !flags.SlackNotificationsDisabled
+
+	default:
+		grip.Alert(message.Fields{
+			"message": "notificationIsEnabled saw unknown subscriber type",
+			"cause":   "programmer error",
+		})
+	}
+
+	return false
+}
+
+type eventMetaJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	env      evergreen.Environment
+	events   []event.EventLogEntry
+	flags    *evergreen.ServiceFlags
+
+	Collection string `bson:"collection" json:"collection" yaml:"collection"`
+}
+
+func makeEventMetaJob() *eventMetaJob {
+	j := &eventMetaJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    eventMetaJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+func NewEventMetaJob(collection string) amboy.Job {
+	j := makeEventMetaJob()
+	j.env = evergreen.GetEnvironment()
+	j.Collection = collection
+
+	// TODO: not safe
+	j.SetID(fmt.Sprintf("%s:%s:%d", eventMetaJobName, time.Now().String(), job.GetNumber()))
+
+	return j
+}
+
+func NewEventMetaJobQueueOperation(collection string) amboy.QueueOperation {
+	return func(queue amboy.Queue) error {
+		err := queue.Put(NewEventMetaJob(collection))
+
+		return errors.Wrap(err, "failed to queue event-metajob")
+	}
+}
+
+func (j *eventMetaJob) fetchEvents() (err error) {
+	j.events, err = event.Find(j.Collection, db.Query(event.UnprocessedEvents()))
+	return err
+}
+
+func (j *eventMetaJob) dispatchLoop() error {
+	// TODO: if this is a perf problem, it could be multithreaded. For now,
+	// we just log time
+	startTime := time.Now()
+	logger := event.NewDBEventLogger(j.Collection)
+	catcher := grip.NewSimpleCatcher()
+	for i := range j.events {
+		var notifications []notification.Notification
+		notifications, err := notification.NotificationsFromEvent(&j.events[i])
+		catcher.Add(err)
+
+		grip.Error(message.WrapError(err, message.Fields{
+			"job_id":     j.ID(),
+			"job":        eventMetaJobName,
+			"source":     "events-processing",
+			"message":    "errors processing triggers for event",
+			"event_id":   j.events[i].ID.Hex(),
+			"event_type": j.events[i].Type(),
+		}))
+
+		if err = notification.InsertMany(notifications...); err != nil {
```

Comment:
I worry that there may be a small number of notifications per event and we'll still have a large number of operations. 

Maybe it makes sense to (and not perhaps as this ticket), but have some sort of buffered insert interface. 

(EVG-3062)

Reply:
discussed earlier, fair point. I've added EVG-3064 for this